packages/app-extensions/src/rest/helpers.js

```diff
@@ -102,14 +102,26 @@ export function* fetchEntities(
 /**
  * Helper to fetch forms.
  * @param formName {String} Name of the requested form
+ * @param allowNotFound {Boolean} If true and the form does not exist null is returned.
+ *                                Otherwise an exception will be thrown.
  */
-export function* fetchForm(formName) {
+export function* fetchForm(formName, allowNotFound = false) {
   const cachedForm = cache.get('form', formName)
   if (cachedForm) {
     return cachedForm
   }
 
-  const response = yield call(requestSaga, `forms/${formName}`)
+  const options = {
+    ...(allowNotFound && {acceptedStatusCodes: [404]})
+  }
+
+  const response = yield call(requestSaga, `forms/${formName}`, options)
+
+  if (allowNotFound && response.status === 404) {
+    cache.add('form', formName, null)
```

Comment:
what's the point of adding it to the cache? as far as I can see, there is no immediate return if `null` is obtained from the cache on line 110. Or should the condition on line be `if (cachedForm || allowNotFound)` to prevent an additional request on line 118 if the form has already been fetched as wasn't found?